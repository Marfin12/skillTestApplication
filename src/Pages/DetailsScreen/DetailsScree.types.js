import * as React from 'react';
import { Image, ScrollView } from 'react-native';

import TextField from '../../Components/TextField';

import styles from './DetailsScreen.styles';

const DetailsScreen = (props) => {
  const data = props.route.params.props;

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.containerScroll}>
      <Image source={{ uri: data.image }} style={styles.photo} />
      <TextField style={styles.description}>
        {data.description}
      </TextField>
    </ScrollView>
  );
};

export default DetailsScreen;
