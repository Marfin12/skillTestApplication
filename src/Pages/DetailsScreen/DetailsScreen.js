import * as React from 'react';
import { View, Image } from 'react-native';

import TextField from '../../Components/TextField';

import styles from './DetailsScreen.styles';

/* eslint-disable react/destructuring-assignment */
const DetailsScreen = (props) => {
  const data = props.route.params.props;

  return (
    <View style={styles.container}>
      <Image source={{ uri: data.image }} style={styles.photo} />
      <TextField style={styles.description}>
        {data.description}
      </TextField>
    </View>
  );
};

export default DetailsScreen;
