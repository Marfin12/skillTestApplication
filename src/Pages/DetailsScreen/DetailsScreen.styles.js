const styles = {
  container: {
    marginLeft: 10
  },
  containerScroll: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  description: {
    fontSize: 11,
    fontStyle: 'italic',
    marginTop: 10,
    width: '80%',
  },
  photo: {
    marginTop: 20,
    height: 200,
    width: '80%',
  },
};

export default styles;
