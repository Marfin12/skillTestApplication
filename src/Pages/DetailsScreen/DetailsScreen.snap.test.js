import React from 'react';
import renderer from 'react-test-renderer';
import DetailsScreen from './DetailsScreen';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => {
        return lightMode;
      })
      .mockImplementationOnce(() => {
        return darkMode;
      }),
  };
});
jest.mock('../../Components/TextField', () => 'TextField');

const route = {
  params: {
    props: {
      name: 'Sport Example',
      image: 'photo of img',
      description: 'some description',
    },
  },
};

test('renders detail screen light mode', () => {
  const ScreenWrapper = renderer
    .create(<DetailsScreen route={route} />)
    .toJSON();
  expect(ScreenWrapper).toMatchSnapshot();
});

test('renders detail screen dark mode', () => {
  const ScreenWrapper = renderer
    .create(<DetailsScreen route={route} />)
    .toJSON();
  expect(ScreenWrapper).toMatchSnapshot();
});
