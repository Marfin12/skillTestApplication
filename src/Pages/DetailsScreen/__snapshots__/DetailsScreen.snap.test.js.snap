// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders detail screen dark mode 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "photo of img",
        }
      }
      style={
        Object {
          "height": 200,
          "marginTop": 20,
          "width": "80%",
        }
      }
    />
    <TextField
      style={
        Object {
          "fontSize": 11,
          "fontStyle": "italic",
          "marginTop": 10,
          "width": "80%",
        }
      }
    >
      some description
    </TextField>
  </View>
</RCTScrollView>
`;

exports[`renders detail screen light mode 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  style={
    Object {
      "marginLeft": 10,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "photo of img",
        }
      }
      style={
        Object {
          "height": 200,
          "marginTop": 20,
          "width": "80%",
        }
      }
    />
    <TextField
      style={
        Object {
          "fontSize": 11,
          "fontStyle": "italic",
          "marginTop": 10,
          "width": "80%",
        }
      }
    >
      some description
    </TextField>
  </View>
</RCTScrollView>
`;
